lib/spack/spack/build_systems/oneapi.py

```diff
@@ -62,24 +62,44 @@ def install(self, spec, prefix, installer_path=None):
             # with other install depends on the userid. For root, we
             # delete the installercache before and after install. For
             # non root we redefine the HOME environment variable.
-            if getpass.getuser() == 'root':
-                shutil.rmtree('/var/intel/installercache', ignore_errors=True)
-
             bash = Executable('bash')
-
-            # Installer writes files in ~/intel set HOME so it goes to prefix
-            bash.add_default_env('HOME', prefix)
             # Installer checks $XDG_RUNTIME_DIR/.bootstrapper_lock_file as well
             bash.add_default_env('XDG_RUNTIME_DIR',
                                  join_path(self.stage.path, 'runtime'))
 
-            bash(installer_path,
-                 '-s', '-a', '-s', '--action', 'install',
-                 '--eula', 'accept',
-                 '--install-dir', prefix)
-
-            if getpass.getuser() == 'root':
-                shutil.rmtree('/var/intel/installercache', ignore_errors=True)
+            if '2021' in str(spec.version):
+                if getpass.getuser() == 'root':
+                    if exists('/var/intel/installercache'):
```

Comment:
1. Let's move `/var/intel/installercache` to a variable.
2. Why not just rename the directory where it is? Like `/var/intel/installercache-backupXXX`. What if `/tmp/intel/` does not exist? What if `/tmp/intel/` is on another partition with limited disk space?